### PR TITLE
Enable CoCoInstanceSegmentationRepresentation in mean_iou metric

### PR DESCRIPTION
**Pull request by mzuevx**
Adds semantic_only flag to MsCocoSegmentationConverter
Implements CoCoInstanceSegmentationRepresentation returns segmentation mask in semantic_only mode
Enables  CoCoInstanceSegmentationAnnotation in resize_segmentation_mask dispatcher
Enables  CoCoInstanceSegmentationAnnotation in mean_iou metrics

These improvements needed to support Deeplab v3 ONNX model, 38764

### CODE REVIEW

File: tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py [hunk 1]
```diff
@@ -262,6 +262,20 @@ class MSCocoSegmentationConverter(MSCocoDetectionConverter):
     __provider__ = 'mscoco_segmentation'
     annotation_types = (CoCoInstanceSegmentationAnnotation, )
 
+    @classmethod
+    def parameters(cls):
+        configuration_parameters = super().parameters()
+        configuration_parameters.update({
+            'semantic_only': BoolField(
```
Comment (eizamaliev) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py:269]: You should update description in `tools/accuracy_checker/accuracy_checker/annotation_converters/README.md`

**Review by eizamaliev — CHANGES_REQUESTED** [review]

**Comment by eaidova** [comment]
@mzuevx  I sent my comments via skype, but repeat here.
You should not use `CoCoInstanceSegmentationAnnotation` in semantic segmention task metrics (mean_iou)
Instead that, converter with key semantic_only should return SegmentationAnnotation which will store already converted polygons of instance mask to semantic label (stored inside of annotation or better saved to the disk and provided path for reducing memory consumption)

**Comment by mzuevx** [comment]
Ok, I'll fix it

File: tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py (hunk 1 above)
Comment (mzuevx) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py:286]: Fixed

**Comment by eaidova** [comment]
Jenknins ok to test

File: tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py [hunk 2]
```diff
@@ -119,17 +133,20 @@ def convert(self, check_content=False, progress_callback=None, progress_interval
...
-        meta.update({'label_map': label_map})
+        self.meta.update({'label_map': label_map})
         detection_annotations, content_errors = self._create_representations(
             image_ids, annotations, label_id_to_label, check_content, progress_callback, progress_interval
         )
 
-        return ConverterReturn(detection_annotations, meta, content_errors)
+        return ConverterReturn(detection_annotations, self.meta, content_errors)
+
+    def meta_update(self, updates):
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py:148]: why do you need this method?
you can directly update meta in your code without this method call

File: tools/accuracy_checker/accuracy_checker/annotation_converters/README.md [hunk 3]
```diff
@@ -101,6 +101,8 @@ Accuracy Checker supports following list of annotation converters and specific f
   * `sort_annotations` - allows to save annotations in a specific order: ascending order of image id or ascending order of image size.
   * `sort_key` - key by which annotations will be sorted(supported keys are `image_id` and `image_size`, default is `image_id`).
   * `dataset_meta_file` - path path to json file with dataset meta (e.g. label_map, color_encoding).Optional, more details in [Customizing dataset meta](#customizing-dataset-meta) section.
+  * `semantic_only` - converts MS COCO dataset annotation to SegmentationAnnotation
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/README.md:104]: please provide default value

File: tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py [hunk 4]
```diff
@@ -270,13 +306,31 @@ def _create_representations(
         num_iterations = len(image_info)
         progress_reporter = PrintProgressReporter(print_interval=progress_interval)
         progress_reporter.reset(num_iterations, 'annotations')
+        segmentation_colors = (
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py:309]: I do not think that you need code classes in color. Just label index will be enough and acceptable (less postprocessing required)

File: tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py [hunk 5]
```diff
@@ -270,13 +306,31 @@ def _create_representations(
...
-            annotation = CoCoInstanceSegmentationAnnotation(image[1], segmentations, image_labels)
+
+            if not self.semantic_only:
+                annotation = CoCoInstanceSegmentationAnnotation(image[1], segmentations, image_labels)
+            else:
+                # print(image_id)
+                h, w, _ = image[2]
+                mask_file = self.masks_dir / "{:012}.png".format(image[0])
+                self.make_mask(h, w, mask_file, image_labels, segmentations, palette)
+                annotation = SegmentationAnnotation(image[1], mask_file, mask_loader=GTMaskLoader.SCIPY)
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py:332]: please, do not forget that serialization of path is platform specific thing. You need to wrap mask_file to string and made relative

Comment (mzuevx) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py]: Fixed

File: tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py (hunk 4 above)
Comment (mzuevx) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py]: Fixed

File: tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py (hunk 2 above)
Comment (mzuevx) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py]: Fixed

File: tools/accuracy_checker/accuracy_checker/annotation_converters/README.md (hunk 3 above)
Comment (mzuevx) [reply tools/accuracy_checker/accuracy_checker/annotation_converters/README.md]: fixed

File: tools/accuracy_checker/accuracy_checker/annotation_converters/README.md [hunk 6]
```diff
@@ -101,6 +101,8 @@ Accuracy Checker supports following list of annotation converters and specific f
   * `sort_annotations` - allows to save annotations in a specific order: ascending order of image id or ascending order of image size.
   * `sort_key` - key by which annotations will be sorted(supported keys are `image_id` and `image_size`, default is `image_id`).
   * `dataset_meta_file` - path path to json file with dataset meta (e.g. label_map, color_encoding).Optional, more details in [Customizing dataset meta](#customizing-dataset-meta) section.
+  * `semantic_only` - converts MS COCO dataset annotation to SegmentationAnnotation. (Optional, default value is False)
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/README.md:104]:
```suggestion
  * `semantic_only` - converts MS COCO dataset annotation to `SegmentationAnnotation`. (Optional, default value is False)
```

File: tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py [hunk 7]
```diff
@@ -104,6 +117,7 @@ def configure(self):
         self.sort_key = self.get_value_from_config('sort_key')
         self.images_dir = self.get_value_from_config('images_dir') or self.annotation_file.parent
         self.dataset_meta = self.get_value_from_config('dataset_meta_file')
+        self.meta = {}
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py:120]: I do not see this field usage. Do we really need this?

File: tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py [hunk 8]
```diff
@@ -276,7 +334,17 @@ def _create_representations(
...
+
+            if not self.semantic_only:
+                annotation = CoCoInstanceSegmentationAnnotation(image[1], segmentations, image_labels)
+            else:
+                h, w, _ = image[2]
+                mask_file = self.masks_dir / "{:012}.png".format(image[0])
+                make_segmentation_mask(h, w, mask_file, image_labels, segmentations)
+                annotation = SegmentationAnnotation(image[1],
+                                                    str(mask_file.relative_to(self.masks_dir.parent)),
+                                                    mask_loader=GTMaskLoader.SCIPY)
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py:346]: scipy uses pillow inside and used for byte manipulation in RGB image in VOC, for grayscale pillow will be enough
```suggestion
                                                    mask_loader=GTMaskLoader.PILLOW)
```

File: tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py [hunk 9]
```diff
@@ -257,11 +271,55 @@ def convert(self, check_content=False, progress_callback=None, progress_interval
...
+                is_directory=True, optional=True,
+                description='path to segmentation masks, used if semantic_only is True'
+            ),
+        })
+        return configuration_parameters
+
+    def configure(self):
+        super().configure()
+        self.semantic_only = self.get_value_from_config('semantic_only')
+        self.masks_dir = self.get_value_from_config('masks_dir')
```
Comment (eaidova) [inline tools/accuracy_checker/accuracy_checker/annotation_converters/ms_coco.py:321]: we need check that mask_dir provided if semantic_only enabled and raise error or define default path for saving (self.annotation_file.parent/'masks').  Also we need create dir if it does not exists in this case.

**Review by eizamaliev — APPROVED** [review]

**Review by eaidova — APPROVED** [review]

**Pull request merged**